components/atom/button/src/index.js

```diff
@@ -55,52 +52,116 @@ const getModifiers = props => {
-  if (color || design) {
-    if (design !== DESIGNS.LINK) {
-      color = color || 'primary'
+function deprecated(
+  validator,
+  callback = (props, propName, componentName) => {
+    console.warn(
+      `The prop ${'\x1b[32m'}${propName}${'\u001b[39m'} is DEPRECATED on ${'\x1b[32m'}${componentName}${'\u001b[39m'}.`
+    )
+  }
+) {
+  return function deprecated(props, propName, componentName, ...rest) {
+    if (props[propName] != null && process.env.NODE_ENV === 'development') {
+      callback(props, propName, componentName, ...rest)
     }
-    design = design || DESIGNS.SOLID
-  } else {
-    type = type || 'primary'
+    return validator(props, propName, componentName, ...rest)
   }
+}
 
-  return {
-    color,
+const typeConversion = ({type, design, color, link, href, ...other}) => {
+  const result = {
     design,
-    type,
-    alignment,
-    ...props
+    color,
+    link,
+    href,
+    ...other
+  }
+  switch (type) {
+    case 'primary':
+      result.color = color || 'primary'
+      result.design = design
+        ? design
+        : link || href
+        ? DESIGNS.LINK
+        : DESIGNS.SOLID
+      break
+    case 'accent':
+      result.color = color || 'accent'
+      result.design = design
+        ? design
+        : link || href
+        ? DESIGNS.LINK
+        : DESIGNS.SOLID
+      break
+    case 'secondary':
+      result.color = color || 'primary'
+      result.design = design
+        ? design
+        : link || href
+        ? DESIGNS.LINK
+        : DESIGNS.OUTLINE
+      break
+    case 'tertiary':
+      result.color = color || 'primary'
+      result.design = design
+        ? design
+        : link || href
+        ? DESIGNS.LINK
+        : DESIGNS.FLAT
+      break
+    default:
+      result.color = color || 'primary'
+      result.design = design
+        ? design
+        : link || href
+        ? DESIGNS.LINK
+        : DESIGNS.SOLID
+      break
   }
+  return result
 }
 
+const getPropsWithDefaultValues = ({
+  type,
+  design,
+  color,
+  alignment,
+  link,
+  href,
+  ...other
+}) => ({
+  ...other,
+  link,
+  type: undefined,
+  design: design ? design : link || href ? DESIGNS.LINK : DESIGNS.SOLID,
+  color: color || 'colors',
+  alignment: alignment || ALIGNMENT.CENTER
+})
+
 const AtomButton = props => {
   const {
-    color,
-    children,
+    alignment,
     className,
+    children,
+    color,
+    design,
+    disabled,
+    isLoading,
     focused,
     groupPosition,
+    link,
     leftIcon,
+    loadingText,
+    loader = <ButtonSpinnerIcon />,
     rightIcon,
     size,
-    design,
-    alignment,
-    title,
-    disabled,
-    isLoading,
-    loadingText,
-    type,
-    loader = <ButtonSpinnerIcon />
-  } = getPropsWithDefaultValues(props)
+    title
+  } = getPropsWithDefaultValues(typeConversion(props))
 
   const classNames = cx(
     CLASS,
-    !type && COLOR_CLASSES[color],
-    !type && CLASSES[design],
-    type && CLASSES[type],
+    COLOR_CLASSES[color],
```

Comment:
Oooone little question, COLOR_CLASSES is till needed then? or the color classes names could be placed in CLASSES directly now since type has disappeared?

Reply:
cierto, mañana lo miro 👍

Reply:
DONE